src/app/project-report/project-report.page.ts

```diff
@@ -0,0 +1,298 @@
+
+@Component({
+  selector: 'app-project-report',
+  templateUrl: './project-report.page.html',
+  styleUrls: ['./project-report.page.scss'],
+})
+export class ProjectReportPage implements OnInit {
+  isFilter: boolean = false;
+  loader: LoaderService;
+  baseApiService: any;
+  listType!: keyof UrlConfig;
+  toastService: ToastService;
+  reportData: any;
+  reportType: string = "1";
+  programId: string = "";
+  isModalOpen = false;
+  selectedProgram: string = "";
+  programList: any;
+  projectsCategories:any;
+   backgroundColors = [
+    'rgb(255, 99, 132)',
+    'rgb(54, 162, 235)',
+    'rgb(255, 205, 86)',
+    'rgb(255, 159, 64)',
+    'rgb(75, 192, 192)',
+    'rgb(255, 99, 132)',
+    'rgb(54, 162, 235)',
+    'rgb(255, 205, 86)',
+  ]
+
+  constructor(
+    private navCtrl: NavController,
+    private router: ActivatedRoute
+  ) {
+    this.loader = inject(LoaderService);
+    this.baseApiService = inject(ApiBaseService);
+    this.toastService = inject(ToastService);
+    Chart.register(...registerables);
+    Chart.register(ChartDataLabels);
+  }
+
+  ngOnInit() {
+    this.listType = 'report';
+    this.getReportData(this.reportType);
+     this.projectsCategories = [
+      {
+        name: 'Total Projects',
+        img: '/assets/images/report-imgs/Note 1.svg',
+        key: 'total',
+      },
+      {
+        name: 'Projects submitted',
+        img: '/assets/images/report-imgs/note.svg',
+        key: 'submitted',
+      },
+      {
+        name: 'Projects inProgress',
+        img: '/assets/images/report-imgs/Note 4.svg',
+        key: 'inProgress',
+      },
+      {
+        name: 'Projects started',
+        img: '/assets/images/report-imgs/Note 3.svg',
+        key: 'started',
+      },
+    ];
+    // this.getPrograms();
+    setTimeout(() => {
+      this.renderChart(this.reportData?.tasks, this.reportData?.categories);
+    });
+  }
+
+  goBack() {
+    this.navCtrl.back();
+  }
+
+  getReportType(e: any) {
+    this.reportType = e;
+    this.getReportData(this.reportType);
+  }
+
+  setOpen(isOpen: boolean) {
+    this.isModalOpen = isOpen;
+  }
+
+  async getReportData(reportType: string) {
+    await this.loader.showLoading('Please wait while loading...');
+    // if (!this.programId) {
+      this.baseApiService.get(urlConfig[this.listType].listingUrl + `?reportType=${reportType}`)
+        .pipe(finalize(async () => {
+          await this.loader.dismissLoading();
+        }))
+        .subscribe((res: any) => {
+          if (res?.status === 200) {
+            if (res.result.dataAvailable) {
+              this.reportData = res.result.data;
+              this.renderChart(this.reportData.tasks, this.reportData.categories); // Update charts after data fetch
+            } else {
+              this.setOpen(true);
+            }
+          }
+        }, (err: any) => {
+          this.toastService.presentToast(err?.error?.message,'danger');
+        });
+    // } else {
```

Comment:
u can remove this if else condition which is commented

Reply:
done